src/include/OSL/rendererservices.h

```diff
@@ -29,6 +30,125 @@ typedef const void* TransformationPtr;
 typedef void (*PrepareClosureFunc)(RendererServices*, int id, void* data);
 typedef void (*SetupClosureFunc)(RendererServices*, int id, void* data);
 
+enum class AttributeSpecBuiltinArg {
+    ShaderGlobalsPointer,  // void* (ideally ShaderGlobals*)
+    ShadeIndex,            // int
+    Derivatives,           // int (ideally bool)
+    Type,                  // long long (ideally TypeDesc)
+    ArrayIndex,            // int, Always zero for non-indexed array lookups.
+    ArrayLookup,           // int (ideally bool)
+    ObjectName,            // const char* (ideally ustring)
```

Comment:
For any string parameters being passed at shader execution, we are trying to move to using ustringhash only, even for CPU/host (to simplify things).  So recommend these parameters be size_t (64bit ustring hashcode pod).  The "RS fmt specification" PR adds a "ustringhash_pod" to be more self documenting.